core/src/main/java/feign/stream/StreamDecoder.java

```diff
@@ -103,4 +123,17 @@ public Type getOwnerType() {
       return null;
     }
   }
+
+  /**
+   * Default implementation of Decodes an http response into an Iterator
+   */
+  public static class LineToIteratorDecoder implements Decoder {
+
+    @Override
+    public Object decode(Response response, Type type) throws IOException {
+      BufferedReader bufferedReader = new BufferedReader(response.body().asReader(UTF_8));
+      return bufferedReader.lines().iterator();
+    }
+
+  }
```

Comment:
For these types of delegate style decoders, we don't typically include a default implementation of a delegate.  I recommend moving this into your test class to act as an example to users on how to provide your own delegate.

The risk here is if a User uses this implementation, we'll have to support it long term.